fcrepo-connector-file/src/test/java/org/fcrepo/connector/file/ExternalJsonSidecarExtraPropertyStoreTest.java

```diff
@@ -54,12 +72,269 @@ public void testSidecarFile() throws IOException {
         when(mockConnector.isContentNode("/file")).thenReturn(false);
         when(mockConnector.fileFor("/file/fcr:content")).thenReturn(new File(FILE_PATH));
         when(mockConnector.isContentNode("/file/fcr:content")).thenReturn(true);
+        when(mockConnector.fileFor("/test/test")).thenReturn(new File(FEDERATION_ROOT + "/test/test"));
+        when(mockConnector.isContentNode("/test/test")).thenReturn(false);
+    }
+
+    @Test
+    public void testSidecarFile() throws IOException {
+        final File tmp = createTempDirectory("filesystem-federation").toFile();
+        tmp.deleteOnExit();
 
         final ExternalJsonSidecarExtraPropertyStore store =
                 new ExternalJsonSidecarExtraPropertyStore(mockConnector, mockTranslator, tmp);
         assertEquals(new File(tmp, "federation-root.modeshape.json"), store.sidecarFile("/"));
         assertEquals(new File(tmp, "file.modeshape.json"), store.sidecarFile("/file"));
         assertEquals(new File(tmp, "file.content.modeshape.json"), store.sidecarFile("/file/fcr:content"));
+        assertEquals(new File(tmp + "/test", "test.modeshape.json"), store.sidecarFile("/test/test"));
+    }
+
+    @Test
+    public void testEmptyPropertyFile() throws IOException {
+        final File tmp = createTempDirectory("filesystem-federation").toFile();
+        tmp.deleteOnExit();
+
+        final ExternalJsonSidecarExtraPropertyStore store =
+                new ExternalJsonSidecarExtraPropertyStore(mockConnector, mockTranslator, tmp);
+
+        assertFalse(store.contains("/file"));
+        assertTrue(store.getProperties("/file").isEmpty());
+    }
+
+    @Test
+    public void testStoreProperties() throws IOException {
+        final File tmp = createTempDirectory("filesystem-federation").toFile();
+        tmp.deleteOnExit();
+
+        final ExternalJsonSidecarExtraPropertyStore store =
+                new ExternalJsonSidecarExtraPropertyStore(mockConnector, mockTranslator, tmp);
+
+        final Map<Name, Property> properties = new HashMap<>();
+
+        store.storeProperties("/file", properties);
+
+        assertFalse(store.contains("/file"));
+        assertTrue(store.getProperties("/file").isEmpty());
+
+        final Name langProp = new BasicName("lang", "de");
+        final Name key1 = new BasicName("info", "one");
+        final Name key2 = new BasicName("info", "two");
+        final Name key3 = new BasicName("info", "three");
+        final Name key4 = new BasicName("info", "four");
+        final Property prop1 = new BasicSingleValueProperty(langProp, "eins");
+        final Property prop2 = new BasicSingleValueProperty(langProp, "zwei");
+        final Property prop3 = new BasicSingleValueProperty(langProp, "drei");
+
+        properties.put(key1, prop1);
+        properties.put(key2, prop2);
+        properties.put(key3, prop3);
+        properties.put(key4, null);
```

Comment:
The thing about the `null` values is that when a property's value is set to `null`, the object corresponding to that key in the backing JSON document is removed. If the value is non-null, the corresponding property is updated. Unless there are objections, I'd be more inclined to keep the implementation as-is, since we know it works at present.